ambry-cloud/src/main/java/com.github.ambry.cloud/azure/AzureCloudDestination.java

```diff
@@ -161,7 +161,12 @@ public boolean updateBlobExpiration(BlobId blobId, long expirationTime) throws C
       return Collections.emptyMap();
     }
 
-    // TODO: For single blob GET request, get metadata from ABS
+    // For single blob GET request, get metadata from ABS instead of Cosmos
+    if (blobIds.size() == 1) {
+      CloudBlobMetadata metadata = azureBlobDataAccessor.getBlobMetadata(blobIds.get(0));
```

Comment:
Why do we want to avoid cosmos here? Because of consistency concern between ABS and cosmos (but that should apply to all cases) OR coz cosmos is more costly?

Reply:
We don't want the extra round trip of going to cosmos (note that with this change there will still be two round trips to ABS, so there is additional work to do).  And it will be expensive to query Cosmos on every GET/Getinfo request.